src/crimson/net/Connection.h

```diff
@@ -34,12 +35,18 @@ class Connection : public boost::intrusive_ref_counter<Connection,
   Connection() {}
   virtual ~Connection() {}
 
+  void set_priv(const SharedPtr& o) {
+    priv = o;
+  }
+  SharedPtr get_priv() {
```

Comment:
does it make sense to templaterize this method by adding a template parameter, like
```c++
template<class T> auto get_priv() {
  return boost:: dynamic_pointer_cast<T>(priv);
}
```
as it's rare that somebody is really interested in a `SharedPtr`.

Reply:
Implemented, but not quite confident...

Because `seastar::enable_shared_from_this` privately inherits `seastar::shared_ptr_count_base`, neither `seastar::dynamic_pointer_cast` nor `seastar::static_pointer_cast` works here.

Investigated a lot, the only solution I found is to use `reinterpret_cast` and return a raw pointer of `T` from `priv.get()`

Reply:
A better one might be to return `reinterpret_cast<T*>(priv.get())->shared_from_this()` instead of raw pointer, what do you think?

Reply:
discussed over IRC, we will drop the `{set,get}_priv()` change. and use a `std::map<>` for tracking the mapping from connection to session instead.